### PR TITLE
Simplify Assignments.identity method

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description
Improve readability

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java [hunk 1]
```diff
@@ -141,19 +141,12 @@ public boolean isIdentity(Symbol output)
...
     {
-        for (Map.Entry<Symbol, Expression> entry : assignments.entrySet()) {
-            Expression expression = entry.getValue();
-            Symbol symbol = entry.getKey();
-            if (!(expression instanceof SymbolReference && ((SymbolReference) expression).getName().equals(symbol.getName()))) {
-                return false;
-            }
-        }
-        return true;
+        return assignments.keySet().stream().allMatch(this::isIdentity);
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java:149]: This way incurs map look-ups while the current code is just iterating over all values

Comment (chenjian2664) [reply core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java:149]: The increase look-ups only when the `value` is `identity` and will call the `map.get`, I think the cost may can be ignore, but improve the readability, WDYT?

Comment (hashhar) [reply core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java:149]: planner is performance sensitive and the older code didn't have bad readability anyway.

Comment (chenjian2664) [reply core/trino-main/src/main/java/io/trino/sql/planner/plan/Assignments.java:149]: Alright, I was confused at the first glance, thanks

**Pull request closed without merging**